Sources/Workspace/Destination.swift

```diff
@@ -126,6 +137,69 @@ public struct Destination {
       #endif
     }
 
+    private static func _hostToolchainSearchPaths() -> [() -> [AbsolutePath]] {
```

Comment:
We should try to use UserToolchain here. Can you look at UserToolchain.swift and see how that can fit in here? Let me know if you need more pointers.

Reply:
> We should try to use UserToolchain here.

I don’t understand. Constructing a `UserToolchain` [requires already having a complete `Destination` instance](https://github.com/SDGGiesbrecht/swift-package-manager/blob/c2042203924089933dfa860910ea7dc3cb36f5f5/Sources/Workspace/UserToolchain.swift#L193), so client code currently looks like this:

```swift
let hostToolchain = try UserToolchain(
    destination: .hostDestination("/wherever/I/think/the/compiler/is."))

// ...then cobble some other things together and eventually acquire a `Manifest` instance.
```

***

Or do you just want the static method moved to the `UserToolchain` namespace?

***

P.S. I was planning on filling in a whole bunch of default values in a follow up pull request, so that clients can just do `Manifest("package/path")` without even thinking about destinations or toolchains. But this pull request had to happen first because the bottom of that stack of defaults is this this `hostBinDir(...)` method, so until it can find itself properly, any default values built on top of it would be broken too.

Reply:
Ya we should keep all lookup code in UserToolchain because it already has a bunch of method for dealing with env, etc.

Reply:
Roger. Shuffling it around now.

Reply:
Everything has been moved to `UserToolchain.swift`.